pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -338,13 +343,111 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
+	}
+}
+
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	nodeInfo := framework.NewNodeInfo(oldPod)
+	if !satisfyVolumeConflicts(pod, nodeInfo) {
+		return framework.QueueSkip
+	}
+
+	return framework.QueueImmediately
+}
+
+// isSchedulableAfterNodeAdded is invoked whenever a node added, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.QueueImmediately
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	originPersistentVolumeClaim, modifiedPersistentVolumeClaim, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
 	}
+
+	usePVC := false
+	// Check if the pod's pvc is created
+	if pl.FilterPVCFromPod(pod, modifiedPersistentVolumeClaim) {
+		usePVC = true
+	}
+
+	if !usePVC {
+		// This was not the PersistentVolumeClaim the pod was waiting for.
+		logger.V(6).Info("unrelated PersistentVolumeClaim got modified", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+		return framework.QueueSkip
+	}
+
+	if !pl.enableReadWriteOncePod {
+		logger.V(7).Info("spec of PersistentVolumeClaim for pod got updated", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+		return framework.QueueImmediately
+	}
+
+	// If the PVC is modified but the accessModes is not modified, the modified has no relate with the pod, and the pod is still in unschedulable
+	if oldObj != nil && apiequality.Semantic.DeepEqual(originPersistentVolumeClaim.Spec.AccessModes, modifiedPersistentVolumeClaim.Spec.AccessModes) {
```

Comment:
I don't think we can change the accessModes of a pvc(can they?). I think we can just check if this pvc is a RWOP pvc and is used by other pods here.

Reply:
Actually, from #103082. I think we added pvc event to check RWOP volumes, so it should be ok to only check RWOPs here.

Reply:
> I don't think we can change the accessModes of a pvc(can they?)

`accessModes` are indeed immutable, have removed `DeepEqual` here